datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```diff
@@ -176,20 +176,62 @@ private fun Questionnaire.QuestionnaireItemComponent.createQuestionnaireResponse
   QuestionnaireResponse.QuestionnaireResponseItemComponent {
   return QuestionnaireResponse.QuestionnaireResponseItemComponent().apply {
     linkId = this@createQuestionnaireResponseItem.linkId
-    if (this@createQuestionnaireResponseItem.initial.isNotEmpty()) {
-      answer =
-        mutableListOf(
-          QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent().apply {
-            value = this@createQuestionnaireResponseItem.initial[0].value
-          }
-        )
-    }
+    answer = createQuestionnaireResponseWithInitialItem()
     this@createQuestionnaireResponseItem.item.forEach {
       this.addItem(it.createQuestionnaireResponseItem())
     }
   }
 }
 
+private fun Questionnaire.QuestionnaireItemComponent.createQuestionnaireResponseWithInitialItem():
```

Comment:
this function name is a bit of a misnomer. it's creating a list of answers rather than the questionnaire response item.

```suggestion
private fun Questionnaire.QuestionnaireItemComponent.createQuestionnaireResponseItemAnswers():
```

and add documentation to this function to say you are using inital items.

Reply:
Done